src/main/java/io/neonbee/data/DataVerticle.java

```diff
@@ -343,10 +343,14 @@ public static <U> Future<U> requestData(Vertx vertx, DataRequest request, DataCo
                         LOGGER.correlateWith(context).debug("Received event bus reply");
 
                         if (asyncReply.succeeded()) {
+                            DataContext responseDataContext =
+                                    decodeContextFromString(asyncReply.result().headers().get(CONTEXT_HEADER));
                             context.setData(Optional
                                     .ofNullable(
                                             decodeContextFromString(asyncReply.result().headers().get(CONTEXT_HEADER)))
                                     .map(DataContext::data).orElse(null));
+                            ((DataContextImpl) context).mergeResponseData(Optional.ofNullable(responseDataContext)
+                                    .map(DataContext::responseData).orElse(null));
```

Comment:
I don't entirely understand this one? as soon as data is received back, we always merge the responseData we received into our own responseData? isn't it that we have to put this into our receivedData instead?! only if propagation is done via the propageteResponseData method then this may be called, isn't it?

Reply:
the data context will be encoded and decoded when tranferred via event-bus. Here we are receiving the decocded data with updates from eventbus.